File: java/org.financer.client/src/main/java/org/financer/client/domain/model/transaction/Attachment.java
package org.financer.client.domain.model.transaction;

import java.time.LocalDate;

public class Attachment {

    private long id;
    private Transaction transaction;
    private String name;
    private LocalDate uploadDate;
    private byte[] content;

    /*
     * Getters and Setters
     */

    public long getId() {
        return id;
    }

    public Attachment setId(long id) {
        this.id = id;
        return this;
    }

    public Transaction getTransaction() {
        return transaction;
    }

    public Attachment setTransaction(Transaction transaction) {
        this.transaction = transaction;
        return this;
    }

    public String getName() {
        return name;
    }

    public Attachment setName(String name) {
        this.name = name;
        return this;
    }

    public LocalDate getUploadDate() {
        return uploadDate;
    }

    public Attachment setUploadDate(LocalDate uploadDate) {
        this.uploadDate = uploadDate;
        return this;
    }

    public byte[] getContent() {
        return content;
    }

    public Attachment setContent(byte[] content) {
        this.content = content;
        return this;
    }
}

File: java/org.financer.client/src/main/kotlin/org/financer/client/domain/model/transaction/Attachment.kt
package org.financer.client.domain.model.transaction

import java.time.LocalDate

class Attachment {
    /*
     * Getters and Setters
     */ var id: Long = 0
        private set
    var transaction: Transaction? = null
        private set
    var name: String? = null
        private set
    var uploadDate: LocalDate? = null
        private set
    lateinit var content: ByteArray
        private set

    fun setId(id: Long): Attachment {
        this.id = id
        return this
    }

    fun setTransaction(transaction: Transaction?): Attachment {
        this.transaction = transaction
        return this
    }

    fun setName(name: String?): Attachment {
        this.name = name
        return this
    }

    fun setUploadDate(uploadDate: LocalDate?): Attachment {
        this.uploadDate = uploadDate
        return this
    }

    fun setContent(content: ByteArray): Attachment {
        this.content = content
        return this
    }
}